### PR TITLE
Support multiple Bosh DNS aliases addon's

### PR DESCRIPTION
**Pull request by rohitsakala**
[#174440477](https://www.pivotaltracker.com/n/projects/2192232/stories/174440477)

### CODE REVIEW

File: pkg/kube/util/boshdns/interface.go [hunk 1]
```diff
@@ -22,14 +22,16 @@ type DomainNameService interface {
...
 func New(m bdm.Manifest) (DomainNameService, error) {
 	dns := NewBoshDomainNameService(m.InstanceGroups)
-
-	index := HasBoshDNSAddOn(m)
-	if index != -1 {
-		if err := dns.Add(m.AddOns[index]); err != nil {
-			return nil, errors.Wrapf(err, "error loading BOSH DNS configuration")
+	for index, addon := range m.AddOns {
+		for _, job := range addon.Jobs {
+			if job.Release == bdm.BoshDNSAddOnName {
```
Comment (manno) [inline pkg/kube/util/boshdns/interface.go:27]: This needs to look at 'bosh-dns' and 'bosh-dns-aliases' I think.

File: pkg/kube/util/boshdns/interface.go [hunk 2]
```diff
@@ -65,8 +67,10 @@ func DNSSetting(m bdm.Manifest, serviceIP, namespace string) (corev1.DNSPolicy,
 func HasBoshDNSAddOn(m bdm.Manifest) int {
 	index := -1
 	for index, addon := range m.AddOns {
-		if addon.Name == bdm.BoshDNSAddOnName {
-			return index
+		for _, job := range addon.Jobs {
+			if job.Release == bdm.BoshDNSAddOnName {
```
Comment (manno) [inline pkg/kube/util/boshdns/interface.go:71]: This too

**Review by manno — CHANGES_REQUESTED** [review]

File: pkg/kube/util/boshdns/interface.go (hunk 1 above)
Comment (rohitsakala) [reply pkg/kube/util/boshdns/interface.go]: Done.

File: pkg/kube/util/boshdns/interface.go (hunk 2 above)
Comment (rohitsakala) [reply pkg/kube/util/boshdns/interface.go]: Done.

**Review by manno — APPROVED** [review]

**Pull request merged**